llvm/lib/Target/AIE/AIE2GenFixupInstrInfo.td

```diff
@@ -760,7 +760,9 @@ let Itinerary = II_VMOV_W in {
                            ItinRegClassPair<II_VMOV_W_WMH_WMH,[OperandRegClass<0, eWH>, OperandRegClass<1, eWH>]>,
                            ItinRegClassPair<II_VMOV_W_WML_WML,[OperandRegClass<0, eWL>, OperandRegClass<1, eWL>]>,
                            ItinRegClassPair<II_VMOV_W_WMH_WML,[OperandRegClass<0, eWH>, OperandRegClass<1, eWL>]>,						   
-                           ItinRegClassPair<II_VMOV_W_WML_WMH,[OperandRegClass<0, eWL>, OperandRegClass<1, eWH>]>] in {	
+                           ItinRegClassPair<II_VMOV_W_WML_WMH,[OperandRegClass<0, eWL>, OperandRegClass<1, eWH>]>,
+                           ItinRegClassPair<II_VMOV_W_WML_Q, [OperandRegClass<0, eWL>, OperandRegClass<1, mQQm>]>,
+                           ItinRegClassPair<II_VMOV_W_WMH_Q, [OperandRegClass<0, eWH>, OperandRegClass<1, mQQm>]>] in {	
 	def VMOV_mv_w : AIE2_mv_w_inst_mv< (outs OP_mMvAMWQDst:$dst), (ins OP_mMvAMWQSrc:$src),
```

Comment:
Could you think of an easy way to check that the `II_VMOV_W` itinerary is never used by the postmisched? Doing so can result in problems like the one you describe. Maybe in `getSchedClass` we should assert if we fallback to the default itinerary because none of the operands matched?

Reply:
is that specific for `II_VMOV_W` or for all the reg-based itinerary ? 

If its specific to II_VMOV_W itinerary I can update the tablegen flow, where in an extra field if marked with an instruction will introduce an assert. Also doing this specifically for `postmisched` will be tricky since we do not have any info in the newly autogenerated `getSchedClass` function. 

Doing it for all is easy but will require adding lot of itinerary manually for MOV instructions since we use conservative itinerary in cases when we are writing to single register like SP, LR , LE .... and so on.

Reply:
I think we should aim to have "fully-covering" reg-based itineraries actually. But yes, I don't know how much work that might be to actually make all of our itineraries complete. I think it's fine for now.